subsys/bluetooth/host/l2cap.c

```diff
@@ -551,11 +551,14 @@ static void l2cap_le_encrypt_change(struct bt_l2cap_chan *chan, uint8_t status)
 #if defined(CONFIG_BT_L2CAP_ECRED)
 	if (chan->ident) {
 		struct bt_l2cap_chan *echan[L2CAP_ECRED_CHAN_MAX];
-		struct bt_l2cap_le_chan *ch;
+		struct bt_l2cap_chan *ch;
 		int i = 0;
 
-		while ((ch = l2cap_remove_ident(chan->conn, chan->ident))) {
-			echan[i++] = &ch->chan;
+		SYS_SLIST_FOR_EACH_CONTAINER(&chan->conn->channels, ch, node) {
+			atomic_clear_bit(ch->status, BT_L2CAP_STATUS_ENCRYPT_PENDING);
```

Comment:
We should only clear the `BT_L2CAP_STATUS_ENCRYPT_PENDING` flag when the `indent` matches and the `ch` is added to `echan`.

Reply:
Yes, that is the wrong place.